core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -759,7 +759,7 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
 
       val future = result.lowWatermarks().get(topicPartition)
       try {
-        lowWatermark = future.get(1000L, TimeUnit.MILLISECONDS).lowWatermark()
+        lowWatermark = future.get().lowWatermark()
```

Comment:
Thanks for looking into this  @soenkeliebau . I agree with your reasoning on the JIRA. But I think as for the fix it is better to still set the timeout which is equal to the default value of `waitUntilTrue` (it is 15secs), because otherwise we may be blocked more than what is specified as the limit in `waitUntilTrue`, and then still catch the `TimeoutException` internally and return false.

cc @cmccabe

Reply:
Where do we catch the `TimeoutException` internally? Using a timeout in `get` is a bit of an anti-pattern, set the request timeout in `AdminClient` appropriately.

Reply:
1. Regarding set the timeout: that's right, as long as we set the timeout config in `adminClient` that should be sufficient, that `future.get()` without parameter would still throw after configured timeout.
2. We should still catch `TimeoutException` in `future.get()` and return false;

Reply:
I've added some code to unnest the LeaderNotAvailableException and rethrow if anything else is nested. Also catch the TimeoutException and return false.
As far as I can tell, timeouts are set to 20 in AdminClient and defaults to 15 for waitUntilTrue, so the Timeout _should_ never occur and we are fine - if it does, we handle it.
I'm not sure if in this case it wouldn't be better to explicitly state the timeouts though, to make the purpose of the test easier to understand without digging through default timeout values. Then again, I could just add a comment to that effect :)  I'll wait for your input before I push more commits..

Reply:
I don't think we want to catch the `TimeoutException` since it's longer than `waitUntilTrue` and it's a different failure mode than if `waitUntilTrue` times out because `LeaderNotAvailableException` happened for too long.

Reply:
Ok, I've removed the check for TimeoutException, since Guozhang hasn't weighed in again and he was advocationg for catching it.